test/seq/GridapTopOptTests/ThermalComplianceHPMTests.jl

```diff
@@ -83,6 +83,16 @@ function main(;order,AD_case)
     PDEConstrainedFunctionals(J,[Vol],state_map,analytic_dJ=dJ)
   elseif AD_case == :partial_ad2
     PDEConstrainedFunctionals(J,[Vol],state_map,analytic_dC=[dVol])
+  elseif AD_case == :custom_pcf
+    objective = GridapTopOpt.StateParamMap(J,state_map)
+    constraints = map(Ci -> GridapTopOpt.StateParamIntegrandWithMeasure(Ci,state_map),[Vol])
+    function φ_to_jc(φ)
```

Comment:
These are great! Can you please create some tests in `seq/StateMapTests`? Perhaps do a similar thing with `AffineFEStateMapTest.jl` and `AffineFEStateMapTest_CutFEM_Elast.jl`. That way we also have the finite difference verification.